TilerServer/TilerWebModule.pas

```diff
@@ -3698,69 +3703,518 @@ function TSliceDiffGeometryICLR.GenerateTileCalc(const aExtent: TExtent; aBitmap
+    else Result := aActiveValue;
+  end
+  else
+  begin
+    if not IsNaN(aRefValue)
+    then Result := aRefValue
+    else aValidFlag := False;
+  end;
+end;
+
+function TSliceDiffGeometryICLR2.GetPaletteColor(aActiveTexture, aRefTexture, aWidth: Double; var aValidFlag: Boolean): TGeoColors;
+begin
+  if not (IsNaN(aActiveTexture) or IsNaN(aRefTexture)) then
+  begin
+    if awidth > 0 then
+      Result := fPalette.ValueToColors(fPalette.minValue())
+    else if awidth < 0 then
+      Result := fPalette.ValueToColors(fPalette.maxValue())
+    else
+      Result := fPalette.ValueToColors((fPalette.minValue() + fPalette.maxValue())/2);
+  end
+  else aValidFlag := False;
+end;
+
+function TSliceDiffGeometryICLR2.ComputeCoordinateDist(aWidth, aActiveValue, aRefValue, aCapacityFactor, aXY_Diff, aPerpDist: Double; aIsCommonPoly: Boolean): Double;
+begin
+  if aWidth > 0 then
+  begin
+    if aIsCommonPoly then
+      Result := aActiveValue*aCapacityFactor*aXY_Diff/aPerpDist
+    else
+      Result := aRefValue*aCapacityFactor*aXY_Diff/aPerpDist;
+  end
+  else
+  begin
+    if aWidth < 0 then
+    begin
+      if aIsCommonPoly then
+        Result := aRefValue*aCapacityFactor*aXY_Diff/aPerpDist
+      else
+        Result := aActiveValue*aCapacityFactor*aXY_Diff/aPerpDist;
+    end
+    else
+      Result := aRefValue*aCapacityFactor*aXY_Diff/aPerpDist;
+  end;
+end;
+
+procedure TSliceDiffGeometryICLR2.DrawFillPolygon(aColor: TGeoColors; aBitmap: FMX.Graphics.TBitmap; aPolygonCommon, aPolygonExtra: TPolygon; xCommon, xExtra: Double);
+begin
+  if aColor.fillColor<>0  then
+  begin
+    if xCommon <> xExtra then
+    begin
+      aBitmap.Canvas.Fill.Color := TAlphaColorRec.Lightsteelblue;
+      aBitmap.Canvas.FillPolygon(aPolygonCommon, 1);
+
+      aBitmap.Canvas.Fill.Color := aColor.fillColor;
+      aBitmap.Canvas.FillPolygon(aPolygonExtra, 1);
+    end
+    else
+    begin
+      aBitmap.Canvas.Fill.Color := TAlphaColorRec.Lightsteelblue;
+      aBitmap.Canvas.FillPolygon(aPolygonCommon, 1);
+    end;
+
+  end;
+  if aColor.outlineColor<>0 then
+  begin
+    if xCommon <> xExtra then
+    begin
+      aBitmap.Canvas.Stroke.Color := TAlphaColorRec.Lightsteelblue;
+      aBitmap.Canvas.DrawPolygon(aPolygonCommon, 1);
+
+      aBitmap.Canvas.Stroke.Color := aColor.outlineColor;
+      aBitmap.Canvas.DrawPolygon(aPolygonExtra, 1);
+    end
+    else
+    begin
+      aBitmap.Canvas.Stroke.Color := TAlphaColorRec.Lightsteelblue;
+      aBitmap.Canvas.DrawPolygon(aPolygonCommon, 1);
+    end;
+  end;
+end;
+
+{ TSliceDiffGeometryICLR3 }
+
+constructor TSliceDiffGeometryICLR3.Create(aLayer: TLayer; aPalette: TWDPalette; aTimeStamp: TDateTime; aCurrentSlice, aRefSlice: TSliceGeometryICLR);
+begin
+  inherited Create(aLayer, aPalette, aTimeStamp, aCurrentSlice, aRefSlice);
+end;
+
+function TSliceDiffGeometryICLR3.GenerateTileCalc(const aExtent: TExtent; aBitmap: FMX.Graphics.TBitmap; aPixelWidth, aPixelHeight: Double): TGenerateTileStatus;

```

Comment:
Please split this method on smaler parts.

Reply:
Done!